opencontainers/struct.py

```diff
@@ -63,42 +79,36 @@ def _is_struct(self, attType=None):
         # We can provide a nested attType to check
         if not attType:
             attType = self.attType
-        try:
-            return (
-                Struct in attType.__bases__
-                or StrStruct in attType.__bases__
-                or IntStruct in attType.__bases__
-            )
-        except:
-            return False
+        return is_struct(attType)
 
     def set(self, value):
         """
         Set a new value, and validate the type. Return true if set
         """
 
-        # First pass, it might be another object to add
-        if self._is_struct():
-            newStruct = self.attType()
-            value = newStruct.load(value)
-
-        # If it's a list with another type
-        elif isinstance(self.attType, list) and self.attType:
-            child = self.attType[0]
-
-            # It's either a nested structure
-            if self._is_struct(child):
-
-                # If we have a list of values, generate them
-                if isinstance(value, list):
-                    values = []
-                    for v in value:
+        if not is_struct(type(value)):
+            # First pass, it might be another object to add
+            if self._is_struct():
+                newStruct = self.attType()
+                value = newStruct.load(value)
+
+            # If it's a list with another type
+            elif isinstance(self.attType, list) and self.attType:
+                child = self.attType[0]
+
+                # It's either a nested structure
+                if is_struct(child):
```

Comment:
This could get hairy if we have multiple levels of nesting. Can you implement this possibly with recursion? Once we've grabbed a child (and we know it's a struct) does it not work to call the function again for that struct?

Reply:
I've added a test for nested Structs.
Recursion is not needed as `.load(value)` will construct nested structs for us.